services/movie.py

```diff
@@ -0,0 +1,55 @@
+from django.db.models import QuerySet
+
+from db.models import Movie, Genre, Actor
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> Movie | QuerySet:
+    queryset = Movie.objects.all()
+
+    if genres_ids and actors_ids:
+        queryset = queryset.filter(
+            genres__id__in=genres_ids,
+            actors__id__in=actors_ids
+        ).distinct()
+
+    if genres_ids and not actors_ids:
+        queryset = queryset.filter(
+            genres__id__in=genres_ids
+        )
+
+    if not genres_ids and actors_ids:
+        queryset = queryset.filter(
+            actors__id__in=actors_ids
+        )
```

Comment:
You can simplify your code
```suggestion

    if genres_ids:
        queryset = queryset.filter(
            genres__id__in=genres_ids
        )

    if actors_ids:
        queryset = queryset.filter(
            actors__id__in=actors_ids
        )
```